### DISCUSSION TITLE
Update chart data

### DISCUSSION THREAD

**Discussion by LorDCristhian** [Syslifters/sysreptor#309] [Q&A]
Good morning community, I would like to know how to make the graph only show vulnerabilities whose status is "open" and that the vulnerability counter does not take into account those vulnerabilities with a status of "solved".

**Comment by LorDCristhian** [comment]
Good morning community, I would like to know how to make the graph only show vulnerabilities whose status is "open" and that the vulnerability counter does not take into account those vulnerabilities with a status of "solved".

![2024-08-20_19-02-31](https://github.com/user-attachments/assets/b5ede184-face-4f0f-ad04-53a4e0948ddc)

**Comment by aronmolnar** [comment]
Hi,
We prepare the `finding_stats` object which we use in the demo reports for creating those figures, e.g. for the bar chart:

```
                data: [
                    finding_stats.count_critical,
                    finding_stats.count_high,
                    finding_stats.count_medium,
                    finding_stats.count_low,
                    finding_stats.count_info
                ],
```

You must replace those data items by your own calculation, for example:

```
                data: [
                    findings.filter(f => f.cvss.level === 'critical' && f.retest_status.value !== 'resolved').length,
                ],
```

This filters out the "resolved" status. You must, however, think about how you treat "partial" or the other statuses.

![image](https://github.com/user-attachments/assets/cd70b999-51a4-4ae9-aeba-a47f763d2ed0)

You can also filter for "open": `findings.filter(f => f.cvss.level === 'critical' && f.retest_status.value === 'open').length`

The same you can do for the sentence above the table:

```
  <p>
    In the course of this penetration test
    <comma-and-join>
      <template #critical v-if="findings.filter(f => f.cvss.level === 'critical' && f.retest_status.value !== 'resolved').length > 0"><strong class="risk-critical">{{ findings.filter(f => f.cvss.level === 'critical' && f.retest_status.value !== 'resolved').length }} Critical</strong></template>
```

In the table, you can also filter for "retest_status" before printing the lines (`<template v-if="finding.retest_status.value !== 'resolved'">`):

```
  <table class="finding-summary-table">
    <thead>
      <tr>
        <th>Vulnerability</th>
        <th align="center">Criticality</th>
        <th v-if="report.is_retest" align="center">Remediation Status</th>
      </tr>
    </thead>
    <tbody>
      <tr v-for="finding in findings">
        <template v-if="finding.retest_status.value !== 'resolved'">
          <td>
            <ref :to="finding.id">{{ finding.title }}</ref>
          </td>
          <td align="center">
            <ref :to="finding.id" :class="'risk-' + finding.cvss.level">{{ lodash.capitalize(finding.cvss.level) }}</ref>
          </td>
          <td v-if="report.is_retest" align="center">
            <ref :to="finding.id" :class="'finding-status-' + finding.retest_status.value">{{ finding.retest_status.label }}</ref>
          </td>
        </template>
      </tr>
    </tbody>
  </table>
```

**Reply by LorDCristhian** [reply]
Thank you so much.
